plaidml/op/op.h

```diff
@@ -5,8 +5,11 @@
 #include <string>
 #include <vector>
 
+#include "llvm/Support/FormatVariadic.h"
+
 #include "plaidml/edsl/edsl.h"
 #include "plaidml/op/ffi.h"
+#include "pmlc/util/logging.h"
```

Comment:
This is another dependency I'd like to avoid for the public API (it's fine for WIP testing/etc).

Reply:
For landed code, should I avoid logging altogether? I had been thinking logging all the parameters of a failed convolution would be a nice feature.

I suppose I could construct a complex exception message with stringstream if that's more appropriate than either of these headers.

Reply:
Alternatively, I could move the default stride/dilation/data_dilation value inference into the oplib, and make logging/exception code there analogous to the code for e.g. autopadding

Reply:
Yes, the more we can push into the oplib itself the better I think. That way all language bindings benefit from it.